### PR TITLE
Decorate standard error.html.twig template

### PR DESCRIPTION
**Pull request by zorn-v**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #22964
| License       | MIT

### CODE REVIEW

**Comment by zorn-v** [comment]
Guys, it just update OLD (check it out) ugly standard twig template. Nothing more.

**Comment by jvasseur** [comment]
I'm :-1: because these are templates that are shown in production environment so we should show something as neutral as possible (they should be overridden on each projects to match your site style).

**Comment by zorn-v** [comment]
You prefer to see "White page with 'Oops' ?"
Some projects no need to define their own error pages. But symfony default is disgrace.

**Comment by sstok** [comment]
At least remove the exception-ghost as this is not really an exception, and non-technical users might be confused about this.

**Comment by zorn-v** [comment]
> At least remove the exception-ghost
Yep, I expected it 😄 
Ghost gives charm. If it were is no difficult to change "exception" to "error" I'd did it of course.
I can try to remove "exception" cloud at all.

**Comment by derrabus** [comment]
> You prefer to see "White page with 'Oops' ?"

Yes, mainly because the framework should not expose itself in prod. That exception ghost and the page layout you chose do exactly that.

Also, that error page is end-user facing. That user probably does not know what an exception is. The fact that Symfony uses exceptions to produce 40x/50x error pages is an implementation detail that the original layout hides. The red color indicates that the user did something wrong, which is not necessarily the case.

> But symfony default is disgrace.

Please try to use a more respectful language.

**Comment by zorn-v** [comment]
> Yes, mainly because the framework should not expose itself in prod.

And users of freamework says. It is UGLY. Just coz dont configure error pages...

**Comment by zorn-v** [comment]
When I show error page (not configured 404) from `silex`, I was asked - "Why it is not in symfony ?"

**Comment by nicolas-grekas** [comment]
> the framework should not expose itself in prod

technically, the framework already exposes itself: the basic page is still something specific to Symfony, if one were to build heuristics as to which code is generating this page.

I like the goal here personnaly.

**Comment by derrabus** [comment]
> the basic page is still something specific to Symfony, if one were to build heuristics

Yes, and even without the error pages, you can probably tell by analyzing runtime behavior like HTTP headers etc. My comment was not security-related.

I'd rather like avoid to use Symfony-branding on end-user facing templates which the exception ghost and the layout with the read header are. Users seeing that page should get the impression that there's something broken with the application they're using. The should not get the impression of a "broken Symfony", imho.

> I like the goal here personnaly.

Agreed. I'm not against changing the default error page in general. I just have concerns regarding the layout proposed in this PR.

**Comment by zorn-v** [comment]
> I'd rather like avoid to use Symfony-branding on end-user

Why ? Is it not cool, knowing that some site using symfony ? )

**Comment by zorn-v** [comment]
BTW. I like 'branding' of symfony.
See what you can take, and that it is better to send to long journey 😄

**Comment by zorn-v** [comment]
> I'd rather like avoid to use Symfony branding
Nah. Is "Oops" white page not "Symfony branding" ? Disgrace ?

**Comment by derrabus** [comment]
> Disgrace ?

Again, I don't think calling the work of someone a "disgrace" is a helpful contribution to a discussion.

**Comment by javiereguiluz** [comment]
I disagree with this proposal too. The reason is that I don't want to make users think that it's OK to not customize the error pages in their apps. That's why our default error pages in prod should look like real and finished error pages. Also, not leaking any detail or information about Symfony is nice ... although some people can now associate the "Oops ..." message to Symfony.

**Comment by nicolas-grekas** [comment]
Either we refuse this proposal, or we fix #22964. Right now we have a mix.

**Comment by zorn-v** [comment]
I remove "exception" cloud, when I be able.

> don't want to make users think that it's OK to not customize the error pages in their apps

And because of this we provide ugly pages (BTW, in SILEX it MUCH BETTER).

**Comment by zorn-v** [comment]
Is it ok, that typehints are removed from constructor for 3.4 compat ?

**Comment by nicolas-grekas** [comment]
Yes it is. Is it required? Tests fail without?

**Comment by zorn-v** [comment]
Nope, but I think I make mistake when used `master` branch as base instead of `3.4`

**Comment by nicolas-grekas** [comment]
This is not a bug fix so master is correct :)

**Comment by zorn-v** [comment]
Yep, this is not bugfix. But WHY users of "old" versions of symfony (coz not php 7.1 on server) should see white page ?

**Comment by nicolas-grekas** [comment]
Because we cannot support everyone. We have policies and they help us keep the maintenance burden handleable.

**Comment by zorn-v** [comment]
I agree. One time step against... and we have anarchy.
Just forgot.

File: src/Symfony/Component/Debug/ExceptionHandler.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ class ExceptionHandler
     private $caughtLength;
     private $fileLinkFormat;
 
-    public function __construct(bool $debug = true, string $charset = null, $fileLinkFormat = null)
+    public function __construct($debug = true, $charset = null, $fileLinkFormat = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/ExceptionHandler.php]: Should be reverted

**Review by nicolas-grekas — APPROVED** [review]
With minor comment.
We must display something, and however raw we may want it, it doesn't have to be ugly.
Also the current situation is inconsistent, with the themed page being displayed in prod also by the Debug component, while the B&W one is when Twig is enabled.

File: src/Symfony/Component/Debug/ExceptionHandler.php (hunk 1 above)
Comment (zorn-v) [reply src/Symfony/Component/Debug/ExceptionHandler.php]: Eeeh. I asked about this )

Comment (zorn-v) [reply src/Symfony/Component/Debug/ExceptionHandler.php]: https://github.com/symfony/symfony/pull/27071#issuecomment-386484021

Comment (zorn-v) [reply src/Symfony/Component/Debug/ExceptionHandler.php]: No problem, BTW

**Comment by zorn-v** [comment]
>We must display something, and however raw we may want it, it doesn't have to be ugly.

EXACTLY. 
https://github.com/symfony/symfony/pull/27071#issuecomment-385465011
I didn't pay much attention before I was asked. WHY I MUST to configure error page if I dont want ?

**Comment by javiereguiluz** [comment]
I'm still 👎 about this proposal. The proposed decoration feels "out of place" for prod pages. The "Symfony ghost" and the rest of elements have been created for the dev pages.

If we want to display a non-ugly error page, we could do something like this, which is totally agnostic and doesn't leak any Symfony design or information:

error-404

error-500

**Comment by javiereguiluz** [comment]
In case you think my last proposal is "too simple" or "not ready for prod", this is Google's 404 error page in production:

google-404-error

**Comment by zorn-v** [comment]
> The "Symfony ghost" and the rest of elements have been created for the dev pages.

Smells like discrimination. Users of my projects not worthy cool error page if I lazy to override it ?

**Comment by andrewmy** [comment]
> Users of my projects not worthy cool error page if I lazy to override it ?

Yep.

**Comment by joshlopes** [comment]
👎  Mostly because:

1. Red is generally a bad colour to give to people that don't understand, they would probably think either the website is a scam / fake / wrong, or it was actually the site you were looking for
2. For us DEVs we can actually associate that template to a dev / test enviromnent if i saw that page in production i would think i had debug enabled or some other env was being used, which would be completely misleading.
3. I prefer white (happy pages) even to show an error - which needs to be imo "naked" and clear to the user it was not part of my website.

**Comment by zorn-v** [comment]
> 👎 Mostly because:

You could override it, no ?

**Comment by zorn-v** [comment]
>I'm happy with @javiereguiluz proposal,

But nobody seems this problem before, yaah.

**Comment by joshlopes** [comment]
@zorn-v please don't misunderstand me/us. We are grateful for your contribution.

AFAIK PR's are reviewed by the community, and i think something that doesn't please the majority, usually the author has to re-consider and take in consideration others opinions. I think the majority, currently, is engaged with the idea (so good job!) but the template used is inadequate. 

> You could override it, no?

This also applies to you, everyone can override... But this PR will change everyone's template (no one would merge something, only to have it overridden on millions of projects...). Not that i would override this template, but for the reasons i stated i won't be happy with this template either. 

> But nobody seems this problem before, yaah.

To be honest, for the reasons above... the blank pages suffice for my needs. I think i can speak for all of us, when i say, we try to avoid errors, not make them look nicer like it is something we will show often. The current blank page, states that there is an error in a way i am sure the user will understand something was off with his request. This PR could fix some issues like sending the user back to homepage.

It's up for the authors of symfony to decide, as they have the last word. But currently, it would be a lot easier just to change it to match the majority expectations. 

If i may ask you, what are the reasons for you to keep this red template? (similar to debug / dev pages) ?

**Comment by zorn-v** [comment]
> To be honest, for the reasons above... the blank pages suffice for my needs. 

Yep. Same for me. But I was asked of my friend wich I learn for symfony. Why is my old project (in silex) have cool error page, but symfony one not.

**Comment by joshlopes** [comment]
@zorn-v I'm sorry but that's not a good reason to keep a red template (re-use the debug / dev template), so we can have a "cool" error page. I think in such special cases you would indeed override the template in your project...

But since you already had all this work, what about doing what @javiereguiluz purposed above? or something similar? that would be great, and if you ask me you would be cool as any sf contributor 😎 , here take yours 🕶

**Comment by javiereguiluz** [comment]
@zorn-v as you can see in the previous comments, most people are in favor of a simpler error page design. This doesn't mean that you are wrong; it's just that we think differently about this.

For this reason I've made some changes in your original pull request to implement this simpler design. This is how it looks:

simple-error-pages

It uses the same font stack as used by modern CSS frameworks and it uses the same red color shade as the "danger" text color of Bootstrap 4, ensuring a right contrast ratio. Thanks!

**Comment by javiereguiluz** [comment]
@zorn-v as you requested, I'm closing this pull request and creating a new one with the latest changes: #27699. Thanks!

**Comment by zorn-v** [comment]
Thanks, guys :smile:

**Pull request closed without merging**